base/loading.jl

```diff
@@ -3416,9 +3415,9 @@ function check_clone_targets(clone_targets)
 end
 
 # Set by FileWatching.__init__()
-global mkpidlock_hook
-global trymkpidlock_hook
-global parse_pidfile_hook
+global mkpidlock_hook::Any
```

Comment:
Out of curiosity, what effect does this have?

Reply:
A global can be given a type later, so if it doesn't have a type yet when the code is compiled we emit a `convert` call to an unknown type.

Reply:
Ideally `global x` would be identical to `global x::Any` (like function arguments) but that might break some use cases, not sure.